packages/grid_client/tests/modules/gateways.test.ts

```diff
@@ -185,38 +204,237 @@ test("TC1237 - Gateways: Expose a VM Over Gateway", async () => {
+  log(gatewayRes);
+
+  //Contracts Assertions
+  expect(gatewayRes.contracts.created).toHaveLength(1);
+  expect(gatewayRes.contracts.updated).toHaveLength(0);
+  expect(gatewayRes.contracts.deleted).toHaveLength(0);
+  expect(gatewayRes.contracts.created[0].contractType.nodeContract.nodeId).toBe(gatewayNodeId);
+
+  gatewayResult = await gridClient.gateway.getObj(gw1.name);
+  log(gatewayResult);
+
+  //Gateway Assertions
+  expect(gatewayResult[0].name).toBe(name + "c");
+  expect(gatewayResult[0].status).toBe("ok");
+  expect(gatewayResult[0].type).toContain("name");
+  expect(gatewayResult[0].domain).toContain(name);
+  expect(gatewayResult[0].tls_passthrough).toBe(tlsPassthrough);
+  expect(gatewayResult[0].backends).toStrictEqual(backends);
+
+  domain = "https://" + gatewayResult[0].domain;
+
+  if (await waitForGateway(domain)) {
+    axios.get(domain).then(res => {
+      log(res.status);
+      log(res.statusText);
+      log(res.data);
+      expect(res.status).toBe(200);
+      expect(res.statusText).toBe("OK");
+      expect(res.data).toContain("Directory listing for /");
+      expect(res.data).toContain("bin/");
+      expect(res.data).toContain("dev/");
+      expect(res.data).toContain("etc/");
+    });
+  } else {
+    throw new Error("Gateway is unreachable after multiple retries");
+  }
+
+  //--------------------IPv6--------------------
+  backends = ["http://[" + result[0].publicIP["ip6"].split("/")[0] + "]:8000"];
+  log(backends);
+
+  //Name Gateway Model
+  const gw2: GatewayNameModel = {
+    name: name + "6",
+    node_id: gatewayNodeId,
+    tls_passthrough: tlsPassthrough,
+    backends: backends,
+  };
+
+  gatewayRes = await gridClient.gateway.deploy_name(gw2);
+  log(gatewayRes);
+
+  //Contracts Assertions
+  expect(gatewayRes.contracts.created).toHaveLength(1);
+  expect(gatewayRes.contracts.updated).toHaveLength(0);
+  expect(gatewayRes.contracts.deleted).toHaveLength(0);
+  expect(gatewayRes.contracts.created[0].contractType.nodeContract.nodeId).toBe(gatewayNodeId);
+
+  gatewayResult = await gridClient.gateway.getObj(gw2.name);
+  log(gatewayResult);
+
+  //Gateway Assertions
+  expect(gatewayResult[0].name).toBe(name + "6");
+  expect(gatewayResult[0].status).toBe("ok");
+  expect(gatewayResult[0].type).toContain("name");
+  expect(gatewayResult[0].domain).toContain(name);
+  expect(gatewayResult[0].tls_passthrough).toBe(tlsPassthrough);
+  expect(gatewayResult[0].backends).toStrictEqual(backends);
+
+  domain = "https://" + gatewayResult[0].domain;
+
+  if (await waitForGateway(domain)) {
+    axios.get(domain).then(res => {
+      log(res.status);
+      log(res.statusText);
+      log(res.data);
+      expect(res.status).toBe(200);
+      expect(res.statusText).toBe("OK");
+      expect(res.data).toContain("Directory listing for /");
+      expect(res.data).toContain("bin/");
+      expect(res.data).toContain("dev/");
+      expect(res.data).toContain("etc/");
```

Comment:
These assertions have been repeated for 5 times. I suggest we exclude it in a function and use it whenver we want to avoid redundancy.